### PR TITLE
Communicating Between OP Mainnet and Ethereum in Solidity

### PR DESCRIPTION
**Pull request by krofax**
**Description**
Updated the Communicating Between OP Mainnet and Ethereum in Solidity Tutorial to use Viem

**Tests**

**Additional context**

**Metadata**

Include a link to any github issues that this may close in the following form:
- Fixes - https://github.com/ethereum-optimism/devrel/issues/261

### CODE REVIEW

File: pages/builders/app-developers/tutorials/cross-dom-solidity.mdx [hunk 1]
```diff
@@ -189,46 +179,32 @@ Start the Node.js REPL with the `node` command.
...
 ```
 
 {<h3>Wait until the message is ready to prove</h3>}
 
-The second step to send messages from L2 to L1 is to prove that the message was sent on L2.
+Next, you will send messages from L2 to L1 is to prove that the message was sent on L2.
 You first need to wait until the message is ready to prove.
 
-```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L23 hash=25a072666b6147f8d8983d8223f045b8
+```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L17-L20 hash=be084e2a13c2cad44f6354feeb2bcbcb
```
Comment (brokewhale) [inline pages/builders/app-developers/tutorials/cross-dom-solidity.mdx:207]: this code isn't complete
image

**Review by cpengilly — APPROVED** [review]

**Comment by cpengilly** [comment]
@krofax approved, but some conflicts and comments left to resolve ❤️

File: pages/builders/app-developers/tutorials/cross-dom-solidity.mdx [hunk 2]
```diff
@@ -240,7 +216,7 @@ Feel free to take a quick break while you wait.
 
 Once the message is ready to be proven, you'll send an L1 transaction to prove that the message was sent on L2.
 
-```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L26 hash=17922abea43b3d379404fedd87422dde
+```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L23-L27 hash=1af494e8b4c1533bc9d1ce981f7ec88c
```
Comment (brokewhale) [inline pages/builders/app-developers/tutorials/cross-dom-solidity.mdx:219]: this too
image

File: pages/builders/app-developers/tutorials/cross-dom-solidity.mdx [hunk 3]
```diff
@@ -253,21 +229,21 @@ On OP Mainnet, this takes 7 days.
 We're currently testing fault proofs on OP Sepolia, so withdrawal times reflect Mainnet times.
 </Callout>
 
-```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L29 hash=45d995aab47ec29afee4bb4577ae9303
+```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L30-L33 hash=ef3598d29f837b59cfb89e7bbbc0645d
```
Comment (brokewhale) [inline pages/builders/app-developers/tutorials/cross-dom-solidity.mdx:232]: this too

image

File: pages/builders/app-developers/tutorials/cross-dom-solidity.mdx [hunk 4]
```diff
@@ -253,21 +229,21 @@ On OP Mainnet, this takes 7 days.
...
 ```
 
 {<h3>Relay the message on L1</h3>}
 
 Once the withdrawal is ready to be relayed you can finally complete the message sending process.
 
-```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L32 hash=b5515811ffcf8b9ada15dea8ae666e44
+```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L36-L40 hash=c3e5c5d15c048c231397aebcb92eef2c
```
Comment (brokewhale) [inline pages/builders/app-developers/tutorials/cross-dom-solidity.mdx:239]: image

File: pages/builders/app-developers/tutorials/cross-dom-solidity.mdx [hunk 5]
```diff
@@ -253,21 +229,21 @@ On OP Mainnet, this takes 7 days.
...
 ```
 
 {<h3>Wait until the message is relayed</h3>}
 
 Now you simply wait until the message is relayed.
 
-```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L35 hash=d6e7f89e929eea0ac3217a6751b7e578
+```js file=<rootDir>/public/tutorials/cross-dom-solidity.js#L43-L46 hash=e9da5a2b27185b0b948ac45b2d42a7f9
```
Comment (brokewhale) [inline pages/builders/app-developers/tutorials/cross-dom-solidity.mdx:246]: image

**Review by brokewhale — COMMENTED** [review]
fix code snippets

**Review by brokewhale — APPROVED** [review]

**Pull request merged**